src/sdam/message_manager.rs

```diff
@@ -65,7 +66,7 @@ impl TopologyMessageManager {
 }
 
 async fn wait_for_notification(receiver: &mut Receiver<()>, timeout: Duration) -> bool {
-    let timeout = Delay::new(timeout);
+    let timeout = RUNTIME.delay_for(timeout);
     let message_received = Box::pin(receiver.recv());
 
     match futures::future::select(timeout, message_received).await {
```

Comment:
I think if we replace this `select` with `RUNTIME.timeout`, we can remove the need for `BoxFuture` on `delay_for` and just make it a regular async method.

Reply:
Updated

Reply:
And now `delay_for` can just be `async`, right?

Reply:
Ah, yep. We may need to wrap the return value in a box depending on how we use it, but no need to do that in this case.